apps/account-api/src/services/siwfV2.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import {
+  generateEncodedSignedRequest,
+  generateAuthenticationUrl,
+  VerifiedPhoneNumberCredential,
+  VerifiedEmailAddressCredential,
+  VerifiedGraphKeyCredential,
+  SiwfCredentialRequest,
+} from '@projectlibertylabs/siwfv2';
+import apiConfig, { IAccountApiConfig } from '#account-api/api.config';
+import blockchainConfig, { IBlockchainConfig } from '#account-lib/blockchain/blockchain.config';
+import { BlockchainService } from '#account-lib/blockchain/blockchain.service';
+import { WalletV2RedirectResponseDto } from '#types/dtos/account/wallet.v2.redirect.response.dto';
+
+@Injectable()
+export class SiwfV2Service {
+  private readonly logger: Logger;
+
+  constructor(
+    @Inject(apiConfig.KEY) private readonly apiConf: IAccountApiConfig,
+    @Inject(blockchainConfig.KEY) private readonly blockchainConf: IBlockchainConfig,
+    private blockchainService: BlockchainService,
+  ) {
+    this.logger = new Logger(this.constructor.name);
+  }
+
+  private static requestedCredentialTypesToFullRequest(requestCredentials: string[]): SiwfCredentialRequest[] {
+    const credentials: SiwfCredentialRequest[] = [];
+    if (
+      requestCredentials.includes('VerifiedEmailAddressCredential') &&
+      requestCredentials.includes('VerifiedPhoneNumberCredential')
+    ) {
+      credentials.push({
+        anyOf: [VerifiedEmailAddressCredential, VerifiedPhoneNumberCredential],
+      });
+    } else if (requestCredentials.includes('VerifiedEmailAddressCredential')) {
+      credentials.push({
+        anyOf: [VerifiedEmailAddressCredential],
+      });
+    } else if (requestCredentials.includes('VerifiedPhoneNumberCredential')) {
+      credentials.push({
+        anyOf: [VerifiedPhoneNumberCredential],
+      });
```

Comment:
Why do we need `anyOf` for the case of just one of these?

Reply:
Good point. In theory either should work. I'll need to check with Peter before changing it, so that will likely be in another PR.